src/brogue/IO.c

```diff
@@ -1405,11 +1453,19 @@ void getCellAppearance(short x, short y, uchar *returnChar, color *returnForeCol
 }
 
 void refreshDungeonCell(short x, short y) {
-    uchar cellChar;
-    brogueAssert(coordinatesAreInMap(x, y));
+    enum displayGlyph cellChar;
     color foreColor, backColor;
+    brogueAssert(coordinatesAreInMap(x, y));
+
     getCellAppearance(x, y, &cellChar, &foreColor, &backColor);
     plotCharWithColor(cellChar, mapToWindowX(x), mapToWindowY(y), &foreColor, &backColor);
+
+    // We use different wall sprites depending on what tile is below, so we need
+    // to refresh the cell above too
+    if (--y >= 0) {
```

Comment:
It's a clever use of --y but, on balance, I'd rather not alter parameters

Reply:
I'm curious why not? I could split the comparison and decrement to make it more readable

Reply:
I tend to prefer leaving variables which are parameters to functions unchanged, i.e. assuming they are implicitly const. Here `--y` decrements `y` which is a parameter and uses it later with a changed value.